src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -406,16 +406,29 @@ public bool ContainsFile(string moniker)
             }
         }
 
-        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath)
+        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath, bool additionalFile)
         {
             lock (_gate)
             {
-                return _documentMonikers.TryGetValue(filePath, out var value)
-                    ? value.document
-                    : null;
+                foreach (var document in _documentMonikers[filePath])
+                {
+                    if (document.Key.IsAdditionalFile == additionalFile)
+                    {
+                        return document;
+                    }
+                }
+
+                return null;
             }
         }
 
+        [Obsolete("Call the overload that specifies additionalFile instead.", error: true)]
+        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath)
+        {
+            // Originally this method would return a host document for either, so preserve that until it is removed.
+            return GetCurrentDocumentFromPath(filePath, additionalFile: false) ?? GetCurrentDocumentFromPath(filePath, additionalFile: true);
+        }
```

Comment:
We can't remove it without TS doing another binary insertion at the same time to remove it.  This allows us to ship 15.3 with TS's existing call, but prevent new calls from being added, and forcing them to adopt to the break ASAP.

Removing is also tracked with https://github.com/dotnet/roslyn/issues/20812.